src/main/java/org/mule/extension/db/internal/domain/connection/datasource/DbDataSourceReferenceConnectionProvider.java

```diff
@@ -49,7 +48,7 @@
 @Alias("data-source")
 @ExternalLib(name = "JDBC Driver", description = "A JDBC driver which supports connecting to the Database",
     nameRegexpMatcher = DRIVER_FILE_NAME_PATTERN, type = JAR, optional = true)
-public class DbDataSourceReferenceConnectionProvider implements ConnectionProvider<DbConnection>, Initialisable, Disposable {
+public class DbDataSourceReferenceConnectionProvider extends DbConnectionProvider {
```

Comment:
I don't see a reason why the client cannot contain a `DatasourceReferenceConnectionProvider` implementation and the connector simply extends it adding this annotations. This class is not exporting anything.